### ISSUE TITLE
types are growing with each hop between repos

### ISSUE THREAD

**Issue by aryairani** [unisonweb/unison#1388]
```
bytes
166 /Users/arya/.unison/v1/terms/#017bv*/type.ub
169 repo1/.unison/v1/terms/#017bv*/type.ub
172 repo2/.unison/v1/terms/#017bv*/type.ub
175 repo3/.unison/v1/terms/#017bv*/type.ub
```

In this particular example, `𝕖111` became `𝕖1111`, `𝕖11111`, `𝕖111111`; `𝕖222` becomes `𝕖222`, `𝕖333`, `𝕖3333`; growing 3 bytes per push generation.

Here's the show of the before and after types.
```
(User "\120150\&111"-1. (User "\120150\&222"-2. (User "\120150". (User "\120150\&333"-3. (User "a". (User "b". (Var User "a" -> (({[Var User "\120150"]} Var User "b"))) -> ({[Var User "\120150\&333"-3]} (Var User "b" -> (({[Var User "\120150"]} (Var User "b" -> (({[Var User "\120150"]} Var User "b")))))) -> ({[Var User "\120150\&222"-2]} Var User "b" -> ({[Var User "\120150\&111"-1]} ##Sequence Var User "a" -> ({[Var User "\120150"]} Var User "b"))))))))))
(User "\120150\&1111"-1. (User "\120150\&2222"-2. (User "\120150". (User "\120150\&3333"-3. (User "a". (User "b". (Var User "a" -> (({[Var User "\120150"]} Var User "b"))) -> ({[Var User "\120150\&3333"-3]} (Var User "b" -> (({[Var User "\120150"]} (Var User "b" -> (({[Var User "\120150"]} Var User "b")))))) -> ({[Var User "\120150\&2222"-2]} Var User "b" -> ({[Var User "\120150\&1111"-1]} ##Sequence Var User "a" -> ({[Var User "\120150"]} Var User "b"))))))))))
```

The grown type is equal to the old one, but it shouldn't be growing.

Here's another example, `base.Set.contains : k -> Set k -> Boolean`; it only grows 1 byte per push generation:
```
(User "\120150". (User "\120150\&111"-1. (User "k". Var User "k" -> ({[Var User "\120150\&111"-1]} #9me9g Var User "k" -> ({[Var User "\120150"]} ##Boolean)))))
(User "\120150". (User "\120150\&1111"-1. (User "k". Var User "k" -> ({[Var User "\120150\&1111"-1]} #9me9g Var User "k" -> ({[Var User "\120150"]} ##Boolean)))))
```

Handy for debugging in ghci:
```haskell
λ>  import qualified Unison.Codebase.Serialization.V1 as V1
λ>  let go h = let typeOfTerm = getFromFile (V1.getType V1.getSymbol (pure Unison.Parser.External)); p = ".unison/v1/terms/#" ++ h ++ "/type.ub" in do Just orig <- typeOfTerm ("/Users/arya/" ++ p); Just pushed <- typeOfTerm ("repo1/" ++ p); traceShowM orig; traceShowM pushed; putStrLn $ if orig == pushed then "they are equal" else "they are not equal"

λ> -- note no '#' here, but the full hash is required
λ> go "bivcubo21g1dv6dvi6og2glj7up42tjmkm32kt8s60c5b91hmmhcvaegmkfs91im8sb8rpsducpdssjiv8t07en6dgs6rv7qj7odtb0"
(User "\120150". (User "\120150\&111"-1. (User "k". Var User "k" -> ({[Var User "\120150\&111"-1]} #9me9g Var User "k" -> ({[Var User "\120150"]} ##Boolean)))))
(User "\120150". (User "\120150\&1111"-1. (User "k". Var User "k" -> ({[Var User "\120150\&1111"-1]} #9me9g Var User "k" -> ({[Var User "\120150"]} ##Boolean)))))
they are equal
```

Handy for debugging in ucm:
```bash
$ for i in seq 1 3; do git init --bare repo$i.git; done

# repeat 3 times or whatever:
$ reunison -- -codebase c init; reunison -- -codebase c; rm -rf c
# pull from current repo and push to the next

$ for i in seq 1 3; do git clone repo$i.git; done
$ wc -b repo*/.unison/v1/terms/#whateverhash/type.ub
```

**Comment by aryairani** [comment]
@pchiusano Definitely seems bad, but not top priority.

**Comment by pchiusano** [comment]
@aryairani good sleuthing. Any ideas?

**Comment by pchiusano** [comment]
@aryairani also, seems like this might be fixed by the new file copy based code syncing (https://github.com/unisonweb/unison/pull/1317), do you know if that's true?

**Comment by aryairani** [comment]
@pchiusano It's avoided by #1317, yes; not fixed.

I am guessing that these definitions have already seen some growth (and that that's why we are starting at `𝕖111`?  maybe from when we moved from `oldbase` to `base`? or something like that) but they would stop growing after a fix (or an avoid).

**Comment by pchiusano** [comment]
Yea totally agree @aryairani we should get to the bottom of it.

UPDATE: Okay, I see what's going on. `V1.putSymbol` is defined as:

```Haskell
putSymbol :: MonadPut m => Symbol -> m ()
putSymbol v@(Symbol id _) = putLength id *> putText (Var.name v)
```

Except that `Var.name` already bakes in the`id`:

```Haskell
name :: Var v => v -> Text
name v = case typeOf v of
  User n -> n <> showid v
```

So with each round trip, the current id gets appended to the name.

Fix is just to define `Var.rawName` or some such which doesn't append the variable id, and then use that in `putSymbol`. PR momentarily.

**Comment by aryairani** [comment]
Great!

**Comment by aryairani** [comment]
Ok — `Symbol` getting to be kinda tangled up with `Var` if `Symbol` has to ask `Var` how to serialize a `Symbol` tho.

**Comment by pchiusano** [comment]
> Ok — Symbol getting to be kinda tangled up with Var if Symbol has to ask Var how to serialize a Symbol tho.

Fixed to have a direct implementation in the #1389 PR
